### PR TITLE
Story/quad 113

### PR DESCRIPTION
**Pull request by PawanMahalle**
Making a pull request with intermediate modifications at the end of sprint.

### CODE REVIEW

File: Quadriga/src/main/webapp/WEB-INF/views/projectCollections/addConceptCollection.jsp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<!--
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/projectCollections/addConceptCollection.jsp]: why did you change this file?

Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/projectCollections/addConceptCollection.jsp]: hm, ok, so you copy pasted it? Why do you copy that file? It doesn't seem to have to do anything with a project blog. Also, why is the folder called "projectCollections"? Aren't you working on a project blog?

**Comment by PawanMahalle** [comment]
story/QUAD-113: Added the functionality to add new project blog entry. Please check and merge with develop.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+@PropertySource(value = "classpath:/user.properties")
+@Controller
+/**
+ * This controller class is responsible for providing views for project blog creation.
+ *
+ * @author Pawan Mahalle
+ *
+ */
+public class PublicBlogController {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java:27]: Please read this and adjust formatting
https://diging.atlassian.net/wiki/pages/viewpage.action?pageId=8781898

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	private IProject getProjectDetails(String name) throws QuadrigaStorageException {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java:31]: unnecessary method, this is just a one liner that you can call directly.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	public String projectblog(@PathVariable("projectUnixName") String projectUnixName, Model model)
+			throws QuadrigaStorageException {
+		IProject project = getProjectDetails(projectUnixName);
+
+		if (project == null) {
+			return "forbidden";
+		}
+
+		// Creating Dummy Object
+		List dummyList = new ArrayList();
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java:55]: This is a generic class/interface and you should specify the type of object you're storing.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+		HashMap<String, String> map = new HashMap<String, String>();
+		map.put("title", "Article: Apple’s iPhone Blunder");
+		map.put("text",
+				"<b> Can the United States <b> government compel Apple to help break into the phone of Syed Rizwan Farook, who, along with his wife Tafsheen Malil, gunned down fourteen innocent people last December at the Inland Regional Center in San Bernardino? That question has sparked fireworks in recent days. The dispute arises because Apple has equipped its new iPhones with encryption settings that erase the data contained on the phone whenever ten false password entries have been made. It was agreed on all sides that only Apple has the technology that might overcome the encryption device. [...]");
+		map.put("date", "February 22, 2016");
+		map.put("author", "Daniel T. Richards ");
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java:61]: You should probably create a class for this rather than using a hashmap to hold the data you want to display on the page.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+		dummyList.add(map);
+
+		/*
+		 * for(int i=1; i<=10; i++) { HashMap <String, String> map = new
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java:66]: Do you still need this code? If not, delete it.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java [hunk 7]
```diff
@@ -47,22 +46,22 @@
 @PropertySource(value = "classpath:/user.properties")
 @Controller
 public class WebsiteProjectController {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:48]: Did you make changes in this class?

File: Quadriga/src/main/webapp/WEB-INF/views/addprojectblog.jsp [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+			.init({
+				selector : '#editable',
+				height : 300,
+				plugins : 'advlist autolink save link image lists charmap print preview',
+				menubar : false,
+				toolbar : 'undo redo | styleselect | bold italic | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link image | print preview media fullpage | forecolor backcolor emoticons | save'
+			});
+</script>
+
+<!-- Action not defined for the form -->
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/addprojectblog.jsp:19]: You probably at least want a heading for the page.

File: Quadriga/src/main/webapp/WEB-INF/views/projectblog.jsp [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+			<p>string (1) : ${fn:escapeXml(string1)}</p>	
+			<p>string (2) : ${fn:escapeXml(string2)}</p>
+    		
+	</div>
+	
+   </c:forEach>
+</c:if>
+
+<c:if test="${empty blockentrylist}">
+You don't own any concept collections.
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/projectblog.jsp:34]: ?

File: Quadriga/src/main/webapp/WEB-INF/views/projectblog.jsp [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+</c:if>
+</div>
+
+<div style="float:right;">
+	<a href="sites/${project.unixName}addprojectblog"><i class="fa fa-plus-circle"></i> Add a new entry</a>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/projectblog.jsp:39]: Isn't there a trailing slash missing after the unixName?

Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/projectblog.jsp:39]: Also, only project admins and owner should be allowed to add a new entry.

File: Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp [hunk 11]
```diff
@@ -1,10 +1,5 @@
...
-<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a
-        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/search">Search</a></li>
-<li ${currentPage == "getprojectstatistics" ? "class=\"active\"" : ""}><a
-        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/statistics">Statistics</a></li>
\ No newline at end of file
+<li ${currentPage == "home" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}">Home</a></li>
+<li ${currentPage == "projectblog" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/projectblog">Project Blog</a></li>
+<li ${currentPage == "networks" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/browsenetworks">Networks</a></li>
+<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/networks">Explore</a></li>
+<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a href="">Search</a></li>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp:5]: You deleted the search url and statistics entry.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+	 * 
+	 * @param projectUnixName
+	 *            The project unix name
+	 * @param model
+	 *            Model
+	 * @return view
+	 * @throws QuadrigaStorageException
+	 */
+	@RequestMapping(value = "sites/{projectUnixName}/sites/addprojectblog", method = RequestMethod.GET)
+	public String addprojectblog(@PathVariable("projectUnixName") String projectUnixName, Model model)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java:89]: Only project admins and owners should be allowed to do this.

**Comment by jdamerow** [comment]
Please reopen this PR when you're done.
